backend/curriculum_tracking/serializers.py

```diff
@@ -679,31 +662,54 @@ class Meta:
     stream_name = serializers.CharField(required=True)
     team_name = serializers.CharField(required=True)
 
-from rest_framework.exceptions import ValidationError
 
+from rest_framework.exceptions import ValidationError
 
 
 class ContentItemAgileWeightSerializer(serializers.ModelSerializer):
     class Meta:
         model = models.ContentItemAgileWeight
-        fields = [
-            "id",
-            "flavour_names",
-            "weight",
-            "content_item",
-        ]
+        fields = ["id", "flavour_names", "weight", "content_item"]
 
     flavour_names = serializers.ListField(CharField)
 
-
-    def save(self,**kwargs):
-        content_item = self.validated_data['content_item']
+    def save(self, **kwargs):
+        content_item = self.validated_data["content_item"]
         available_flavours = content_item.flavour_names
-        flavour_names = self.initial_data.getlist('flavour_names')
+        flavour_names = self.initial_data.getlist("flavour_names")
         for flavour in flavour_names:
             if flavour not in available_flavours:
-                raise ValidationError(f"flavour '{flavour}' not allowed. Choose from {available_flavours}")
-        instance = super(ContentItemAgileWeightSerializer,self).save()
+                raise ValidationError(
+                    f"flavour '{flavour}' not allowed. Choose from {available_flavours}"
+                )
+        instance = super(ContentItemAgileWeightSerializer, self).save()
         instance.set_flavours(flavour_names)
         return instance
 
+
+class CourseRegistrationSerialiser(serializers.ModelSerializer):
+
+    user_name = serializers.CharField(required=False)
+    curriculum_name = serializers.CharField(required=False)
+
+    class Meta:
+        model = models.CourseRegistration
+        fields = ["id", "user", "curriculum", "user_name", "curriculum_name"]
+
+    user = serializers.SerializerMethodField("get_user")
+    user_name = serializers.SerializerMethodField("get_user_email")
```

Comment:
Is it a user name or a user email? 

Can you please name stuff according to what it is?

Reply:
Same comment as before. 

You are getting a user's email and calling it a name. Please call an email an email. 

Call a spade a spade. Call an email an email.

You are saying: `user_name = email things`
It would be better to say `user_email = email things`